### PR TITLE
Pawn valid move plus a bunch of tests

### PR DESCRIPTION
**Pull request by jeffp2007**

### CODE REVIEW

**Comment by jeffp2007** [comment]
Why do you think this failed? All tests passed locally.

File: test/models/pawn_test.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    pawn.update_attributes(:pos_y => 3)
+    move = pawn.valid_move?([0, 2])
+    assert_not move
+  end
+
+  test "check if move is valid (invalid case, horizontal move)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    pawn.update_attributes(:pos_y => 2)
+    move = pawn.valid_move?([1, 2)
```
Comment (cryogenx) [inline test/models/pawn_test.rb]: looks like you forgot a bracket ]

**Comment by johnnyiller** [comment]
Usually when tests pass locally but fail on a build server it is because you have data on your local machine that isn't on the remote machine.  Try clearing your DB on local then running again and see if the test still pass locally.

File: test/models/pawn_test.rb (hunk 1 above)
Comment (jeffp2007) [inline test/models/pawn_test.rb]: Yeah, that did the trick. Thanks!!

## 

Jeff Powers
Global Partners, Inc.
www.globalpartnersinc.com

Connect on Linkedin: http://www.linkedin.com/in/jeffreytpowers/

**Pull request closed without merging**